### PR TITLE
Update observation YAMLs for conventional surface pressure

### PR DESCRIPTION
**Pull request by emilyhcliu**
This PR includes the following updates for conventional surface pressure
1. Add observation YAML under configuration
2. Update observation YAML under testing
3. Add conv_ps to 3dvar obs list
4. Set window shift to true (time boundary is inclusive)
5. Validate 3dvar for surface (station) pressure (end-to-end)
     - [QC validation](https://github.com/NOAA-EMC/GDASApp/pull/803#issuecomment-1851387894)
     - [Increments](https://github.com/NOAA-EMC/GDASApp/pull/803#issuecomment-1852496937)

See [PR #792 ](https://github.com/NOAA-EMC/GDASApp/pull/792) for the BUFR converter work to convert the following surface pressure observation into IODA:
- SFC - 181, 187
- SFCSHIP - 180, 183
- SONDE - 120

The related UFO evaluation  (using GSI obs and geovals) results for surface pressure are documented [here](https://docs.google.com/presentation/d/1OvVMYOutxVyk0GE93Jzoh3hQVoj-dCgrwsVWDDlPT1s/edit?usp=sharing).

### CODE REVIEW

**Comment by emilyhcliu** [comment]
### There are two major error inflation procedures for station pressure:
1. error inflation based on duplicate observations at the same lat/lon as a function of observation time
2. error inflation based on discrepancy between observation and model 

We should get the identical initial error and the inflated error for duplicate obs since they rely on observation information only
Differences in final observation errors and number of observation passed QC is expected due to differences in model components  (geovals) between GSI and JEDI

### Here is the **initial observation error** between GSI and JEDI (identical result expected):

Obs Pressure vs. Initial Obs Errors
GSI (red) vs. JEDI (blue) --- they are on top of each other and therefore purple  (blue+red --> purple)
obserr0_hist_noqc

Obs Pressure vs. Initial Obs Errors 
GSI (red) vs. JEDI (blue) --- they are on top of each other
obserr0_vs_ps_scatter_noqc

Here is the **error inflation factor** for duplicate observations between GSI and JEDI (identical result expected)
GSI (red) vs. JEDI (blue) ---> on top of each other ---> purple
dupfact_hist_qc
Obs Pressure vs. Error Inflation Factor for Duplicate Obs
dupfact_vs_ps_scatter_qc
[**Side notes** ](https://github.com/NOAA-EMC/GDASApp/pull/792#issuecomment-1852442578)about the precision of dateTime values

### Here is **final observation error** and **data count passed QC**

GSI (red) vs. JEDI (blue)
obserr_hist_qc

Obs Pressure vs. Final Obs Error
obserr_vs_ps_scatter_qc

Note: Some of the error inflation differences from the SfcPressure check are larger.  This will be investigated in the future.

The total number  of observation passed QC for GSI is 85378
The total number of observation passed QC for JEDI is 85387

They are close enough!

**Comment by emilyhcliu** [comment]
Here are the 3dvar increments for station pressure  --- look reasonable
```
CostJb: FG-BG
----------------------------------------------------------------------------------------------------
Increment print | number of fields = 8 | cube sphere face size: C768
eastward_wind                                | Min:+0.000000e+00 Max:+0.000000e+00 RMS:+0.000000e+00
northward_wind                               | Min:+0.000000e+00 Max:+0.000000e+00 RMS:+0.000000e+00
air_temperature                              | Min:+0.000000e+00 Max:+0.000000e+00 RMS:+0.000000e+00
surface_pressure                             | Min:-1.480315e-01 Max:+1.245782e-01 RMS:+6.496403e-04
specific_humidity                            | Min:+0.000000e+00 Max:+0.000000e+00 RMS:+0.000000e+00
cloud_liquid_ice                             | Min:+0.000000e+00 Max:+1.618770e-20 RMS:+1.293217e-23
cloud_liquid_water                           | Min:+0.000000e+00 Max:+1.474788e-19 RMS:+2.167418e-22
ozone_mass_mixing_ratio                      | Min:+0.000000e+00 Max:+0.000000e+00 RMS:+0.000000e+00
```

File: parm/atm/variational/3dvar_dripcg.yaml [hunk 1]
```diff
@@ -2,6 +2,7 @@ cost function:
   cost type: 3D-Var
   window begin: '{{ ATM_WINDOW_BEGIN | to_isotime }}'
   window length: $(ATM_WINDOW_LENGTH)
+  window shift: true
```
Comment (CoryMartin-NOAA) [inline parm/atm/variational/3dvar_dripcg.yaml]: This syntax has changed in develop this week. Just noting that here.

Comment (emilyhcliu) [reply parm/atm/variational/3dvar_dripcg.yaml]: Noted!!  Thank you.  

current configuration:
```
window begin: "2018-04-14T23:30:00Z"
window end: "2018-04-15T03:00:00Z"
window shift: true
```

new configuration:
```
time window:
  begin: "2018-04-14T23:30:00Z"
  end: "2018-04-15T03:00:00Z"
  bound to include: begin
```

Looks like we will have to update the window section in the YAMLs when we merge develop branches from JCSDA.

Comment (CoryMartin-NOAA) [reply parm/atm/variational/3dvar_dripcg.yaml]: It's already in GDASApp develop. We need to discuss about when to merge this sprint branch into develop of GDASApp. I would suggest doing so as soon as possible. We should chat with @ADCollard and @RussTreadon-NOAA

Comment (ADCollard) [reply parm/atm/variational/3dvar_dripcg.yaml]: > It's already in GDASApp develop. We need to discuss about when to merge this sprint branch into develop of GDASApp. I would suggest doing so as soon as possible. We should chat with @ADCollard and @RussTreadon-NOAA

@CoryMartin-NOAA Is there any reason not to do this?

**Review by ADCollard — APPROVED** [review]

File: parm/atm/variational/3dvar_dripcg.yaml (hunk 1 above)
Comment (CoryMartin-NOAA) [reply parm/atm/variational/3dvar_dripcg.yaml]: Russ and I want to, we just don't want to if there is a reason not to. The sprint can continue, we were just wanting to get what is working into develop "soon"

**Review by nicholasesposito — APPROVED** [review]

**Comment by emilyhcliu** [comment]
Re-ran the station pressure assimilation case using the updated G-W and GDASApp with updated JEDI repositories in lower resolution (C384).  

The filter results stay the same and are comparable to GSI results
```
20004   0: QC surface_ps airTemperature: 163912 missing values.
20005   0: QC surface_ps airTemperature: 8509 rejected as processed but not assimilated.
20007   0: QC surface_ps stationElevation: 172421 rejected as processed but not assimilated.
20008   0: QC surface_ps stationElevation: 0 passed out of 172421 observations.
20009   0: QC surface_ps stationPressure: 79556 missing values.
20010   0: QC surface_ps stationPressure: 6336 black-listed.
20011   0: QC surface_ps stationPressure: 1142 rejected by first-guess check.
20012   0: QC surface_ps stationPressure: 85387 passed out of 172421 observations.
```
Ps increments look reasonable.
```
19730   0: ----------------------------------------------------------------------------------------------------
19731   0: Increment print | number of fields = 8 | cube sphere face size: C384
19732   0: eastward_wind                                | Min:+0.000000e+00 Max:+0.000000e+00 RMS:+0.000000e+00
19733   0: northward_wind                               | Min:+0.000000e+00 Max:+0.000000e+00 RMS:+0.000000e+00
19734   0: air_temperature                              | Min:+0.000000e+00 Max:+0.000000e+00 RMS:+0.000000e+00
19735   0: surface_pressure                             | Min:-1.676442e-01 Max:+1.328963e-01 RMS:+1.334842e-03
19736   0: specific_humidity                            | Min:+0.000000e+00 Max:+0.000000e+00 RMS:+0.000000e+00
19737   0: cloud_liquid_ice                             | Min:+0.000000e+00 Max:+0.000000e+00 RMS:+0.000000e+00
19738   0: cloud_liquid_water                           | Min:+0.000000e+00 Max:+0.000000e+00 RMS:+0.000000e+00
19739   0: ozone_mass_mixing_ratio                      | Min:+0.000000e+00 Max:+0.000000e+00 RMS:+0.000000e+00
19740   0: ----------------------------------------------------------------------------------------------------
```

I think this PR is ready to merge as soon as the [PR #792](https://github.com/NOAA-EMC/GDASApp/pull/792) is approved.

**Comment by emilyhcliu** [comment]
Side notes: I also re-ran test for IASI (with updated G-W and GDASApp, and C384). 
With the same (default) resource in the configuration for `atmanl`,  the iasi run failed with OOM.  It can run to completion after I changed memory from 90 GB to 160 GB.  It took 34 minutes to finish.

**Comment by CoryMartin-NOAA** [comment]
I just merged this, but realized it went into feature/gdas-validation. @RussTreadon-NOAA @emilyhcliu do we want it in develop instead? (sorry for jumping the gun!)

**Comment by emilyhcliu** [comment]
> I just merged this, but realized it went into feature/gdas-validation. @RussTreadon-NOAA @emilyhcliu do we want it in develop instead? (sorry for jumping the gun!)

Let's merge it into develop as well.

**Pull request merged**